pkg/server/registry/apigroups/acorn/jobs/translator.go

```diff
@@ -0,0 +1,134 @@
+package jobs
+
+import (
+	"context"
+	"strings"
+
+	"github.com/acorn-io/baaah/pkg/router"
+	mtypes "github.com/acorn-io/mink/pkg/types"
+	apiv1 "github.com/acorn-io/runtime/pkg/apis/api.acorn.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	"github.com/acorn-io/runtime/pkg/namespace"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apiserver/pkg/storage"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Translator struct {
+	client kclient.Client
+}
+
+func (t *Translator) FromPublicName(ctx context.Context, namespace, name string) (string, string, error) {
+	parts := strings.Split(name, ".")
```

Comment:
There's a helper method to split: https://github.com/acorn-io/runtime/blob/85cfe8c01b57609ae5213adbed04c9e2611e4fb8/pkg/publicname/publicname.go#L23

It return appName, resourceName.  If the string has no `.` in it it will return appName as "".